stix2/sources/filters.py

```diff
@@ -155,7 +116,53 @@ def apply_common_filters(stix_objs, query):
             yield stix_obj
 
 
-"""Base type filters"""
+def _check_filter(filter_, stix_obj):
+    """Evaluate a single filter against a single STIX 2.0 object.
+
+    Args:
+        filter_ (Filter): filter to match against
+        stix_obj: STIX object to apply the filter to
+
+    Returns:
+        True if the stix_obj matches the filter,
+        False if not.
+
+    """
+    if "." in filter_.field:
+        # For properties like granular_markings and external_references
+        # need to extract the first property from the string.
+        field = filter_.field.split(".")[0]
+    else:
+        field = filter_.field
+
+    if field not in stix_obj.keys():
+        # check filter "field" is in STIX object - if cant be
+        # applied to STIX object, STIX object is discarded
+        # (i.e. did not make it through the filter)
+        return False
+
+    if "." in filter_.field:
+        # Check embedded properties, from e.g. granular_markings or external_references
+        sub_field = filter_.field.split(".", 1)[1]
+        sub_filter = filter_._replace(field=sub_field)
+        if isinstance(stix_obj[field], list):
+            for elem in stix_obj[field]:
+                r = _check_filter(sub_filter, elem)
```

Comment:
Since `_check_filter` returns a boolean, this might be clearer as:
```python
if _check_filter(sub_filter, elem) is True:
    return True
```

Reply:
I was thinking we couldn't check if `is True` because `_all_filter` returns -1 if the filter operation is not supported, but I think it would be better to have `_all_filter` raise an error instead of returning the "-1" special value.